test/e2e/parallel/banned_user_test.go

```diff
@@ -0,0 +1,78 @@
+package parallel
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	testspace "github.com/codeready-toolchain/toolchain-common/pkg/test/space"
+	. "github.com/codeready-toolchain/toolchain-e2e/testsupport"
+	testsupportspace "github.com/codeready-toolchain/toolchain-e2e/testsupport/space"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/wait"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestBannedUser(t *testing.T) {
+	t.Parallel()
+
+	// make sure everything is ready before running the actual tests
+	awaitilities := WaitForDeployments(t)
+	hostAwait := awaitilities.Host()
+	memberAwait := awaitilities.Member1()
+
+	// create an user and a space
+	sp, us, _ := testsupportspace.CreateSpaceWithRoleSignupResult(t, awaitilities, "admin",
+		testspace.WithSpecTargetCluster(memberAwait.ClusterName),
+		testspace.WithTierName("appstudio"),
+		testspace.WithStatusProvisionedNamespaces([]toolchainv1alpha1.SpaceNamespace{
```

Comment:
do we need this condition ? I think no, since initially it will be empty and later we wait at line 39 for the provisioned namespaces to be filled.

Reply:
I don't remember why we needed but we can try without it and revert if required. Let me remove this line

Reply:
done in b6d91539da97de88797fe1994abe954ef9c39a60

Reply:
it passed